ext/TrixiConvexECOSExt.jl

```diff
@@ -0,0 +1,158 @@
+# Package extension for adding Convex-based features to Trixi.jl
+module TrixiConvexECOSExt
+
+# Required for coefficient optimization in P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using Convex: MOI, solve!, Variable, minimize, evaluate
+    using ECOS
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..Convex: MOI, solve!, Variable, minimize, evaluate
+    using ..ECOS
+end
+
+# Use other necessary libraries
+using LinearAlgebra: eigvals
+
+# Use functions that are to be extended and additional symbols that are not exported
+using Trixi: Trixi, undo_normalization!, bisect_stability_polynomial, @muladd
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# Undo normalization of stability polynomial coefficients by index factorial
+# relative to consistency order.
+function Trixi.undo_normalization!(consistency_order, num_stage_evals, gamma_opt)
+    for k in (consistency_order + 1):num_stage_evals
+        gamma_opt[k - consistency_order] = gamma_opt[k - consistency_order] /
+                                           factorial(k)
+    end
+    return gamma_opt
+end
```

Comment:
Sorry if I have missed this in my initial review, but the "*main modified argument comes first*" (see the Trixi.jl [docs](https://trixi-framework.github.io/Trixi.jl/stable/styleguide/#Conventions)):
```suggestion
function Trixi.undo_normalization!(gamma_opt, consistency_order, num_stage_evals)
    for k in (consistency_order + 1):num_stage_evals
        gamma_opt[k - consistency_order] = gamma_opt[k - consistency_order] /
                                           factorial(k)
    end
    return gamma_opt
end
```
IMHO this convention is not just for fun, but it actually helps to intuitively use these functions correctly